handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -429,6 +429,10 @@ public void handlerRemoved0(ChannelHandlerContext ctx) throws Exception {
             // Check if queue is not empty first because create a new ChannelException is expensive
             pendingUnencryptedWrites.removeAndFailAll(new ChannelException("Pending write on removal of SslHandler"));
         }
+        if (engine instanceof OpenSslEngine) {
```

Comment:
I wouldn't say its a leak as shutdown() will be called in the `finalize()` as well. It just release memory as fast as possible as you never know when the finalize() method will be run at the end.